### PR TITLE
Feat/be signin

### PR DESCRIPTION
**Pull request by stephenabv**
## Description

_Provide a description on what was changed/added/fixed in this PR_

## Checklist

_First, create a draft pull request. Then mark the PR as ready for review when all necessary checkbox items has been completed_

- [ ] Reviewed own code
- [ ] Commented on code that is hard to understand
- [ ] Implemented tests for the feature/bugfix
- [ ] All **_Required_** GitHub status checks pass
- [ ] The frontend and backend PR previews have been deployed

### CODE REVIEW

File: backend/src/controller/loginController.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if (password !== PASSWORD) {
+      return res.status(401).json({ status: 401, message: unauthorized });
+    }
+
+    // Check if either email or username is present and valid
+    const isUsernameValid = username === USER_NAME;
+    const isEmailValid = email === EMAIL;
+
+    if (isUsernameValid || isEmailValid) {
+      return res.status(200).json({ status: 200, message: message });
```
Comment (edzhabs) [inline backend/src/controller/loginController.ts]: refer to this [PR in signup](https://github.com/Biniverse/projectbiniverse/pull/38) in the additional notes section for verifying the hashed password.

**Review by edzhabs — DISMISSED** [review]

File: backend/src/controller/loginController.ts (hunk 1 above)
Comment (stephenabv) [reply backend/src/controller/loginController.ts]: This is resolved. Thanks

**Review by stephenabv — COMMENTED** [review]
All changes have been implemented

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
   "editor.formatOnSave": true,
-  "editor.defaultFormatter": "esbenp.prettier-vscode"
-}
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "[sql]": {
```
Comment (beefysalad) [inline .vscode/settings.json:4]: can you check if these are valid changes

File: backend/src/index.ts [hunk 3]
```diff
@@ -15,6 +16,14 @@ app.use(helmet());
 app.use(cors());
 app.use(generalLimiter);
 app.use(express.json());
+app.use(
+  session({
+    secret: "secretnisiya",
```
Comment (beefysalad) [inline backend/src/index.ts]: add secret key to .env file

File: backend/src/controller/loginController.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        contact: authenticated.contact,
+        role: authenticated.role,
+      };
+
+      // TODO: Add encryption to encrypt user data that will be used for sessionVerification
+
+      return res.status(200).json({
+        success: true,
+        message: `Login successful ${authenticated.firstName} ${authenticated.lastName}`,
+        user: user,
```
Comment (beefysalad) [inline backend/src/controller/loginController.ts:48]: also, dont forget to pass the jwtoken

File: backend/src/service/loginService.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+): Promise<IUser | null> {
+  try {
+    let user: IUser = await getUserByEmail(email);
+
+    if (!user) {
+      return null;
+    }
+
+    if (user) {
+      console.log(user);
```
Comment (beefysalad) [inline backend/src/service/loginService.ts]: dont forget to remove the console.log after final push

File: backend/src/controller/loginController.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      const user = {
+        employeeId: authenticated.employeeId,
+        firstName: authenticated.firstName,
+        lastName: authenticated.lastName,
+        email: authenticated.email,
+        contact: authenticated.contact,
+        role: authenticated.role,
+      };
+
+      // TODO: Add encryption to encrypt user data that will be used for sessionVerification
```
Comment (beefysalad) [inline backend/src/controller/loginController.ts]: i think for this you can use the jwt.sign method (passing the id of the user (not the employeeId))
see details below for reference

```javascript
//seperate function (could be helper)
const generateToken = (id) => {
  return jwt.sign({ id }, process.env.JWT_SECRET, {
    expiresIn: '30d',
  });
};

return res.status(200).json({
        success: true,
        message: `Login successful ${authenticated.firstName} ${authenticated.lastName}`,
        user: user,
        token:generateToken(id)
        })
```

Comment (beefysalad) [reply backend/src/controller/loginController.ts]: other than that, everything else looks good

File: backend/src/shared/generateToken.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ * The function `generateToken` creates a JWT token with a payload containing a user ID and a specified
+ * expiration time.
+ * @param {any} user - The `user` parameter in the `generateToken` function is typically the user
+ * object or user ID for whom the token is being generated. It is used to create a payload for the
+ * token that typically includes information about the user, such as their ID or any other relevant
+ * data.
+ * @returns The `generateToken` function returns a JSON Web Token (JWT) generated using the `jwt.sign`
+ * method from the `jsonwebtoken` library.
+ */
+export const generateToken = (user: any) => {
```
Comment (beefysalad) [inline backend/src/shared/generateToken.ts]: Dont use :any

**Review by beefysalad — APPROVED** [review]
LOOKS GOOD TO ME 🚀

**Pull request merged**